GitUI/UserControls/RevisionGrid/Graph/GraphModel.cs

```diff
@@ -288,7 +289,17 @@ bool GetOrCreateNode(ObjectId objectId, out Node n)
                 if (!_nodeByObjectId.TryGetValue(objectId, out n))
                 {
                     n = new Node(objectId);
-                    _nodeByObjectId.Add(objectId, n);
+                    if (!_nodeByObjectId.TryAdd(objectId, n))
+                    {
+                        if (Debugger.IsAttached)
+                        {
+                            Debugger.Break();
+                        }
```

Comment:
Just read the documentation, you are right. It can only fail if:
- and argument exception occurs
- overflow exception occurs
- when the key already exists

I removed it.